### DISCUSSION TITLE
Using CMake with -DSTATIC

### DISCUSSION THREAD

**Discussion by paresy** [baresip/baresip#2063] [General]
I know that CMake support is quite new, but i really like it and wanted to jump on the new shiny feature. I want to build baresip as a static library with a few modules and expected to get a libbaresip.a which includes all modules statically linked (as it was before using plain makefiles). Instead it seems that CMake is building a lot of separate libraries which i need to link individually. e.g. modules/ice/ice.a

Is that assumption correct? Or should we get a libbaresip.a that includes all modules? 
If we do not want to build a big libbaresip.a we still might want to consider "installing" all submodules. At the moment i get only get them from the source directory.

Thanks for your input on this :-)

**Comment by sreimers** [comment]
Can you try git main, I just refactored the module STATIC handling: 999501ee8091bc4b9ac82328be95bae36f4f03e3

**Reply by paresy** [reply]
Awesome! Thanks. Works like a charm 😄
